src/python/pants/backend/common/targets/app.py

```diff
@@ -0,0 +1,259 @@
+# coding=utf-8
```

Comment:
I don't love `pants.backend.common` for this, as `pants.backend` is where backends go, not base classes useful to backends. Right now every dir under `src/python/pants/backend` represents a backend, and this would no longer be true with this change.

I really think this should live under `pants.build_graph`. We already have language-neutral target-y things there, like `resources.py`.